internal/provider/metric_export_prometheus_config_resource.go

```diff
@@ -0,0 +1,405 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"net/http"
+	"strings"
+)
+
+var metricExportPrometheusConfigAttributes = map[string]schema.Attribute{
+	"id": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "Cluster ID.",
+		PlanModifiers: []planmodifier.String{
+			stringplanmodifier.RequiresReplace(),
+		},
+	},
+	"enabled": schema.BoolAttribute{
+		Default:     booldefault.StaticBool(true),
+		Optional:    true,
+		Computed:    true,
```

Comment:
Is having both of these allowed? If you're not allowed to set the field, I think it should be `Computed: true` only.  Also, I'm a little confused about the description here.  Can we remove the additional bit about refraining to set the value if we just make it Computed?

Reply:
I was just reading about this.  I see its required because there is a default value.  But I think in this case you should still just use computed with no default since this is synthesized from the status attribute which is also computed.   Or am I not understanding this correctly.  Is this supposed to be a like an on / off switch?

Reply:
We have removed this now. We introduced it to compute diff on the resource. tf plan will show diff in fields if and only if there is change in input fields. Since enable prometheus export doesn't have body, we had added this field to track the integrations status.
Scenario:
- create resource (enable prometheus metric export) through tf.
- delete prometheus metric export integration through API.
- create resource (enable prometheus metric export) through tf.
Earlier this was showing no changes. Now, we are removing the resource if integration is deactivated through API.